local-development/devtool/adapters/node_detector.py

```
@@ -0,0 +1,46 @@
+    """Discovers the Node.js interpreter on the local filesystem."""
+
+    def check_node(self, min_major: int) -> tuple[str, str]:
+        path = shutil.which("node")
+        if not path:
+            raise RuntimeError(
+                f"Node.js not found on PATH. Install Node.js {min_major}+ "
+                f"(https://nodejs.org) or use nvm: 'nvm install {min_major}'"
+            )
+
+        try:
+            out = subprocess.check_output(
```

Ruff (0.15.14)

[error] `subprocess` call: check for execution of untrusted input

(S603)